libraries/standard/mqtt/src/mqtt_state.c

```diff
@@ -113,6 +145,65 @@ static void updateRecord( MQTTPubAckInfo_t * records,
                           MQTTPublishState_t newState,
                           bool shouldDelete );
 
+/**
+ * @brief Get the packet ID and index of a publish in a specified state.
+ *
+ * @note The API is limited to search states of only outgoing publish records
+ * or incoming publish records at a time. The usage of API only demands the
+ * search of only either of the records. Hence this is an optimization to
+ * limit the search only to one of the records at a time.
```

Comment:
You lost me.

Reply:
This comment need to be reworked. Thanks